<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>621</width>
    <height>470</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>621</width>
    <height>470</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>621</width>
    <height>470</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0" colspan="4">
     <widget class="QTextBrowser" name="infoBox">
      <property name="styleSheet">
       <string notr="true">background-color: rgba(255, 255, 255, 203);</string>
      </property>
      <property name="verticalScrollBarPolicy">
       <enum>Qt::ScrollBarAlwaysOn</enum>
      </property>
      <property name="horizontalScrollBarPolicy">
       <enum>Qt::ScrollBarAsNeeded</enum>
      </property>
      <property name="sizeAdjustPolicy">
       <enum>QAbstractScrollArea::AdjustIgnored</enum>
      </property>
      <property name="lineWrapMode">
       <enum>QTextEdit::FixedPixelWidth</enum>
      </property>
      <property name="lineWrapColumnOrWidth">
       <number>582</number>
      </property>
      <property name="html">
       <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'.SF NS Text'; font-size:13pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p align=&quot;center&quot; style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Helvetica,sans-serif'; font-size:18pt; font-weight:600; text-decoration: underline;&quot;&gt;MOSViz Table Generator for JWST/NIRSpec MSA Programs&lt;/span&gt;  &lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;MOSViz is a quick-look analysis and visualization tool for multi-object spectroscopy (MOS). It is designed to work with JWST and non-JWST data: spectra and it can be used with associated images. MOSViz uses a special catalog called a MOSViz Table to organize file paths and information. For JWST data, the MOSViz table generator uses information in the headers of NIRSpec spectra files to generate a MOSViz Table. There is a separate tool to create a table for non-JWST data. &lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-family:'.SF NS Text,sans-serif';&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'.SF NS Text,sans-serif'; font-weight:600; text-decoration: underline;&quot;&gt;Inputs&lt;/span&gt;&lt;span style=&quot; font-family:'.SF NS Text,sans-serif'; text-decoration: underline;&quot;&gt; &lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'.SF NS Text,sans-serif'; font-weight:600;&quot;&gt;NIRSpec spectra directory:&lt;/span&gt;&lt;span style=&quot; font-family:'.SF NS Text,sans-serif';&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'Calibri,sans-serif';&quot;&gt;This is the path to the NIRSpec MSA post-pipeline spectra. These files are saved in the level3 subdirectory of the pipeline output. The files should follow the naming scheme: &lt;/span&gt;&lt;/p&gt;
&lt;p align=&quot;center&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'.SF NS Text,sans-serif'; font-style:italic;&quot;&gt;&amp;lt;programName&amp;gt;_&amp;lt;objectName&amp;gt;_&amp;lt;instrument_filter&amp;gt;_ &amp;lt;grating&amp;gt;_&amp;lt;s2d|x1d&amp;gt;.fits &lt;/span&gt;&lt;/p&gt;
&lt;p align=&quot;center&quot; style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-family:'.SF NS Text,sans-serif'; font-style:italic;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'.SF NS Text,sans-serif'; font-weight:600;&quot;&gt;Cutout Image directory&lt;/span&gt;&lt;span style=&quot; font-family:'.SF NS Text,sans-serif';&quot;&gt;: &lt;/span&gt;&lt;span style=&quot; font-family:'Calibri,sans-serif';&quot;&gt;MOSViz can display a “postage stamp” image of the source along with the spectra. These images are usually cutouts from larger images. You can add cutout images by selecting “Add cutouts” and by providing a path to a directory containing them. If the you would like to generate cutouts from an image, please use the NIRSpec Cutout Tool. You can create and use your own cutouts as well; the cutout images should follow the naming scheme: &lt;/span&gt;&lt;/p&gt;
&lt;p align=&quot;center&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Times New Roman'; font-size:7pt;&quot;&gt; &lt;/span&gt;&lt;span style=&quot; font-family:'.SF NS Text,sans-serif'; font-style:italic;&quot;&gt;&amp;lt;objectName&amp;gt;_&amp;lt;programName&amp;gt;_cutout.fits &lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Times New Roman,serif';&quot;&gt; &lt;/span&gt; &lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'.SF NS Text,sans-serif'; font-weight:600; text-decoration: underline;&quot;&gt;Outputs&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'.SF NS Text,sans-serif'; font-weight:600;&quot;&gt;MOSViz Table&lt;/span&gt;&lt;span style=&quot; font-family:'.SF NS Text,sans-serif';&quot;&gt;: &lt;/span&gt;MOSViz Table: A table containing paths to MSA spectra and cutout images will be saved in the same directory as the spectra files. You can import this file into glue and visualize your data using MOSViz. &lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Information contained in MOSViz Table: &lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;- Object ID. &lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;- Right Ascension. &lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;- Declination. &lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;- Path to 1d spectrum file. &lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;- Path to 2d spectrum file. &lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;- Path to cutout file, “None” otherwise. &lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;- Slit width and length. &lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;- Pixel scale. &lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
     </widget>
    </item>
    <item row="7" column="3">
     <widget class="QPushButton" name="imageBrowseButton">
      <property name="toolTip">
       <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-family:'.SF NS Text,sans-serif';&quot;&gt;You can add cutout images by selecting “Add cutouts” and by providing a path to a directory containing them. The cutout images should follow the following naming scheme: &lt;/span&gt;&lt;span style=&quot; font-family:'.SF NS Text,sans-serif'; font-style:italic;&quot;&gt;&amp;lt;objectName&amp;gt;_&amp;lt;programName&amp;gt;_cutout.fits &lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
      <property name="text">
       <string>Browse</string>
      </property>
     </widget>
    </item>
    <item row="3" column="3">
     <widget class="QPushButton" name="specBrowseButton">
      <property name="toolTip">
       <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Path to level 3 directory containing the NIRSpec fits files. &lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
      <property name="text">
       <string>Browse</string>
      </property>
     </widget>
    </item>
    <item row="3" column="0">
     <widget class="QLabel" name="label">
      <property name="text">
       <string>  Path to spectra directory:</string>
      </property>
     </widget>
    </item>
    <item row="4" column="0" colspan="4">
     <widget class="Line" name="line">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item row="2" column="0" colspan="4">
     <widget class="Line" name="line_3">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item row="3" column="1" colspan="2">
     <widget class="QLineEdit" name="inSpectra"/>
    </item>
    <item row="7" column="1" colspan="2">
     <widget class="QLineEdit" name="inImage">
      <property name="toolTip">
       <string/>
      </property>
     </widget>
    </item>
    <item row="7" column="0">
     <widget class="QLabel" name="label_2">
      <property name="text">
       <string>  Cutout Image directory:</string>
      </property>
     </widget>
    </item>
    <item row="22" column="3">
     <widget class="QPushButton" name="genTableButton">
      <property name="styleSheet">
       <string notr="true">
</string>
      </property>
      <property name="text">
       <string>Generate Table</string>
      </property>
      <property name="autoExclusive">
       <bool>false</bool>
      </property>
      <property name="autoDefault">
       <bool>false</bool>
      </property>
      <property name="default">
       <bool>true</bool>
      </property>
      <property name="flat">
       <bool>false</bool>
      </property>
     </widget>
    </item>
    <item row="21" column="0" colspan="4">
     <widget class="Line" name="line_2">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item row="5" column="1">
     <widget class="QRadioButton" name="addCutoutsRadioButton">
      <property name="toolTip">
       <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-family:'.SF NS Text,sans-serif';&quot;&gt;The cutout images should follow the following naming scheme: &lt;/span&gt;&lt;span style=&quot; font-family:'.SF NS Text,sans-serif'; font-style:italic;&quot;&gt;&amp;lt;objectName&amp;gt;_&amp;lt;programName&amp;gt;_cutout.fits &lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
      <property name="text">
       <string>Add Cutouts </string>
      </property>
     </widget>
    </item>
    <item row="5" column="2">
     <widget class="QRadioButton" name="noCutoutsRadioButton">
      <property name="toolTip">
       <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-family:'.SF NS Text,sans-serif';&quot;&gt;MOSViz can display a “postage stamp” image of the source along with the spectra. If the you would like to automatically generate cutouts from an image, please use the &lt;/span&gt;&lt;span style=&quot; font-family:'.SF NS Text,sans-serif'; font-weight:600; color:#800080;&quot;&gt;NIRSpec Cutout Tool &lt;/span&gt;&lt;span style=&quot; font-family:'.SF NS Text,sans-serif';&quot;&gt;instead to generate the MOSViz Table. &lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
      <property name="text">
       <string>No Cutouts</string>
      </property>
      <property name="checkable">
       <bool>true</bool>
      </property>
      <property name="checked">
       <bool>false</bool>
      </property>
     </widget>
    </item>
    <item row="5" column="0">
     <widget class="QLabel" name="label_3">
      <property name="text">
       <string>  Cutout Images:</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>621</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
